### ISSUE TITLE
Show alternative build menu versions of craftable items on crafting screen along with their crafting ingredients.

### ISSUE THREAD

**Issue by ifffrt** [CleverRaven/Cataclysm-DDA#48786]
# Is your feature request related to a problem? Please describe.
I just discovered that the metal smoking rack from the crafting menu has an alternative, far easier to build version on the build menu. This could have saved me an entire in-game day of breaking down furniture trying to look for sets of pipe fittings just to build the portable version of it.

# Describe the solution you'd like
We should show players whether the item they're trying to craft from the crafting menu have an alternative version they can build instead, as build menu versions of such items frequently requires much more common ingredients. This should ideally include items that are slightly different in some ways but serve fundamentally the same functions, such as showing the forge from the build menu as an alternative version to the craftable electric forge, even though the former uses coals as fuel and the later uses battery. Furthermore, it should also directly show players the crafting ingredients required of said build menu items so players can conveniently compare/contrast those items.

# Describe alternatives you've considered
None.

# Additional context

**Comment by Salty-Panda** [comment]
The Furniture category is very messy
![image](https://user-images.githubusercontent.com/26396019/117533405-f0154300-afec-11eb-83ff-a45e2b454ae1.png)
Furniture without any function (stools, armchairs) is scattered over both Furniture and Construction.

The categories could be based more on player's needs, so there is no need to scroll through 15 types of walls in order to discover there is an option to build Fire Rings, Butchering Racks, etc.

Could be separated into 
Walls, Doors, Windows, Utilities (things that have some utility role like Fire Ring, Mill, Water Well), Furniture (useless stools, etc.)

Basically, split current "Constructions" and "Furniture" into smaller chunks and throw useless stuff into "Decorative"

**Comment by Night-Pryanik** [comment]
Closing as stale, since stalebot can't do this by itself.
